### ISSUE TITLE
Transformer: Trim whitespace in body text nodes

### ISSUE THREAD

**Issue by schlessera** [ampproject/amp-wp#4456]
## Feature description

Add logic to the existing `Minify` transformer so that it collapses and trims whitespace in body text nodes

Related #4213 

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Acceptance criteria

* Generated HTML output from the Optimizer is minified
* The text nodes within `<body>` only contain the essential whitespace to maintain their displayed result

## Implementation brief

* 

## QA testing instructions

1. 

## Demo

* 

## Changelog entry

*

**Comment by westonruter** [comment]
Is this not a duplicate of #4454?

**Comment by schlessera** [comment]
> Is this not a duplicate of #4454?

No, this is about trimming whitespace within the `->textContent` of a node.

**Comment by westonruter** [comment]
How is that not the same as:

> collapse whitespace (except in `<pre>`, `<textarea>`, `<script>`)

**Comment by schlessera** [comment]
#4454:
```html
<div>
   <ul>
      <li>something</li>
   </ul>
</div>
```
=>
```html
<div><ul><li>something</li></ul></div>
```

#4456:
```html
<p>      Some
      text that is wrapped
    across multiple
      lines      </p>
```
=>
```html
<p> Some text that is wrapped across multiple lines </p>
```
Note that the complexity is a different one for text nodes, as some whitespace can be removed while similar whitespace in a different position might have an impact on rendering.
